src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,80 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.RegistrationFailed;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int CORRECT_AGE = 19;
+    private static final String CORRECT_LOGIN = "testlogin";
+    private static final String CORRECT_PASSWORD = "testpassword";
+    private static final String SAME_TEST_LOGIN = "testlogin123";
+    private RegistrationServiceImpl registrationService;
+    private User testUser;
+    private User user1;
+    private User user2;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        testUser = new User();
+        user1 = new User();
+        user2 = new User();
+    }
+
+    @Test
+    void userWithLoginNull_isNotOk() {
+        testUser.setLogin(null);
+        assertThrows(RegistrationFailed.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void userWithPasswordNull_isNotOk() {
+        testUser.setPassword(null);
+        assertThrows(RegistrationFailed.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void userWithAgeNull_isNotOk() {
+        testUser.setAge(null);
+        assertThrows(RegistrationFailed.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void correctLoginLenght() {
+        testUser.setLogin("321");
+        assertThrows(RegistrationFailed.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void correctPassword() {
+        testUser.setPassword("321");
+        assertThrows(RegistrationFailed.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void correctAge() {
+        testUser.setAge(1);
+        assertThrows(RegistrationFailed.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void registrationWithTheSameLogin_IsNotOk() {
```

Comment:
The test `registrationWithTheSameLogin_IsNotOk` is using the `register` method to add the user for testing. As per the checklist, you should use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` to add users for testing.